### PR TITLE
AP_ESC_Telem: use send_struct mavlink function

### PR DESCRIPTION
**Pull request by tridge**
saves some flash, cpu and stack to use the structure method instead of send method
saves 1080 bytes on durandal, 348 on MatekF405
update: added static_assert to check if msgs have been extended

### CODE REVIEW

**Review by andyp1per — APPROVED** [review]
Nice!

**Review by amilcarlucas — APPROVED** [review]
looks great!

File: libraries/AP_ESC_Telem/AP_ESC_Telem.cpp [hunk 1]
```diff
@@ -299,54 +294,63 @@ void AP_ESC_Telem::send_esc_telemetry_mavlink(uint8_t mav_chan)
...
                 break;
             case 7:
-                fn = mavlink_msg_esc_telemetry_29_to_32_send;
+                fn = (esc_telem_send_struct_fn_t)mavlink_msg_esc_telemetry_29_to_32_send_struct;
                 break;
 #endif
         }
         if (fn != nullptr) {
-            fn((mavlink_channel_t)mav_chan, temperature, voltage, current, current_tot, rpm, count);
+            fn((mavlink_channel_t)mav_chan, &s);
```
Comment (khancyr) [inline libraries/AP_ESC_Telem/AP_ESC_Telem.cpp:353]: at this point, won't it be simplier to call the functions directly from the switch case ? the function pointer looks a bit overcomplicated for a simple call with 2 params

Comment (tridge) [reply libraries/AP_ESC_Telem/AP_ESC_Telem.cpp:353]: it is less flash to do it this way

**Pull request merged**